<launch>
  <master auto="start"/>

  <!-- Run the RVIZ -->
  <arg name="model" />
  <arg name="gui" default="false" />
  <param name="robot_description" textfile="$(find volcano_robotsim)/rviz/model.urdf" />
  <param name="use_gui" value="$(arg gui)" />
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find volcano_robotsim)/rviz/volcano_teb.rviz" />
  
  <!-- Gmapping建图算法，接收传感器数据建立地图 -->
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen" >
    <param name="odom_frame" value="odom" />
    <param name="base_frame" value="base_link" />
    <param name="map_update_interval" value="0.25" />
    <!-- Set maxUrange < actual maximum range of the Laser -->
    <param name="maxRange" value="10" />
    <param name="maxUrange" value="9.5" />
    <param name="sigma" value="0.05" />
    <param name="kernelSize" value="1" />
    <param name="lstep" value="0.05" />
    <param name="astep" value="0.05" />
    <param name="iterations" value="5" />
    <param name="lsigma" value="0.075" />
    <param name="ogain" value="3.0" />
    <param name="lskip" value="0" />
    <param name="srr" value="0.01" />
    <param name="srt" value="0.02" />
    <param name="str" value="0.01" />
    <param name="stt" value="0.02" />
    <param name="linearUpdate" value="0.5" />
    <param name="angularUpdate" value="0.157" />
    <param name="temporalUpdate" value="-1.0" />
    <param name="resampleThreshold" value="0.5" />
    <param name="particles" value="50" />

    <param name="xmin" value="-1.0" />
    <param name="ymin" value="-1.0" />
    <param name="xmax" value="1.0" />
    <param name="ymax" value="1.0" />

    <param name="delta" value="0.05" />
    <param name="llsamplerange" value="0.01" />
    <param name="llsamplestep" value="0.01" />
    <param name="lasamplerange" value="0.005" />
    <param name="lasamplestep" value="0.005" />
    <remap from="scan" to="/volcano/Sick_LMS_291/laser_scan/layer0" />
  </node>


  <include file="$(find volcano_robotsim)/launch/move_base_teb.launch"/>

  
</launch>